crates/bevy_ecs/src/observer/mod.rs

```diff
@@ -427,6 +432,41 @@ impl World {
     }
 }
 
+/// Returns an [`Observer`] that watches for an input event, maps it to a new event type, and triggers
+/// that event.
+///
+/// # Examples
+///
+/// ```
+/// # use bevy_ecs::prelude::*;
+/// # use bevy_ecs::observer::trigger_map;
+/// # let mut world = World::new();
+///
+/// #[derive(Event, Clone, Debug)]
+/// struct Foo(usize);
+///
+/// #[derive(Event, Debug)]
+/// struct Bar(bool);
+///
+/// #[derive(Event, Debug)]
```

Comment:
```suggestion
/// /// The combined event to be emitted.
/// ///
/// /// If we don't care about tracking why the event was triggered, we can simply use a unit struct here instead.
/// #[derive(Event, Debug)]
```

Reply:
Some docs for your doc test 😅  Maybe that would be better as a second example?

Reply:
> If we don't care about tracking why the event was triggered, we can simply use a unit struct here instead.

Yeah rather than introducing new / unrelated context into this example, I think a new example that illustrates "attaching info" makes more sense

Reply:
I'm also not sure "the combined event to be emitted" is necessary here. I think `Combined` and `derive(Event)` , in combination with the context of the example is more than enough to be self-describing. Verges on "comments for comment's sake" for my taste. Just uses up space + mental bandwidth I think.

Reply:
> Yeah rather than introducing new / unrelated context into this example, I think a new example that illustrates "attaching info" makes more sense

Ahh realizing now you weren't talking about attaching info, but rather emitting a single shared unit struct. Sadly this won't be quite as trivial as swapping it out because the unit struct isn't a function that matches the expected signature. Need to manually define the mapping function in that case. Just added the example!